src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -144,6 +144,37 @@ public IEnumerable<int> GetPath(int id, UmbracoEntityTypes type)
             return foundContent.Path.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse);
         }
 
+
+
+
+        public dynamic GetUrl(int id, UmbracoEntityTypes type)
+        {
+            dynamic result = new System.Dynamic.ExpandoObject();
+            
+
+            if(type == UmbracoEntityTypes.Document)
+            {
+                var foundUrl = Umbraco.Url(id);
+                if (string.IsNullOrEmpty(foundUrl) == false && foundUrl != "#")
+                {
+                    result.url = foundUrl;
+                    return result;
+                }   
+            }
+
+            var ancestors = GetAncestors(id, type);
+
+            //if content, skip the first node for replicating NiceUrl defaults
+            if(type == UmbracoEntityTypes.Document) {

```

Comment:
Why do we skip the first ancestor here @madsrasmussen & @perploug 

What if I have a site with many top level root nodes with or without domains assigned, would I loose that information in the path for the preview under the picked node?